examples/persistence_redis.ipynb

```diff
@@ -0,0 +1,758 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "51466c8d-8ce4-4b3d-be4e-18fdbeda5f53",
+   "metadata": {},
+   "source": [
+    "# How to create a custom checkpointer using Redis\n",
+    "\n",
+    "When creating LangGraph agents, you can also set them up so that they persist their state. This allows you to do things like interact with an agent multiple times and have it remember previous interactions.\n",
+    "\n",
+    "This example shows how to use `Redis` as the backend for persisting checkpoint state.\n",
+    "\n",
+    "NOTE: this is just an example implementation. You can implement your own checkpointer using a different database or modify this one as long as it conforms to the `BaseCheckpointSaver` interface."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0aac2830",
+   "metadata": {},
+   "source": [
+    "## Install the necessary libraries for Redis on Python"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "faadfb1b-cebe-4dcf-82fd-34044c380bc4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%capture --no-stderr\n",
+    "%pip install -U redis langgraph"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0c384ca8",
+   "metadata": {},
+   "source": [
+    "### 1. Installing Docker\n",
```

Comment:
could you please remove these instructions and just mention that a user will need redis server running on port X

Reply:
Added @vbarda I have removed the install instructions for docker. Does the latest change work?

Reply:
that's much better, thanks! one small comment -- any chance we could use localhost instead? it might be nonintuitive to some users as to why we're using `172.25.0.4`

Reply:
Sure, I can do that! I'll remove the output of the cells if that's fine with you?

Reply:
actually i'd rather we have outputs of the cells!

Reply:
@vbarda I won't be able to test it using `localhost` as I have tested it on the link that I hosted on docker and I have multiple redis instances being used on different applications. So it'd  be tough

Reply:
okay that's fine then! maybe let's just aadd a comment to the cell where you initialize redis client to say something like "if you're running locally, make sure to change this to localhost"?

Reply:
@vbarda Replaced the IP with `localhost` in the example. There's only a few minor merge conflicts which I do not have access to